### DISCUSSION TITLE
linq2db.CLI tool usage

### DISCUSSION THREAD

**Discussion by febus** [linq2db/linq2db#3672] [Q&A] [answered]
Hello,
I'm trying to generate a model classes for my existing database using linq2db CLI tool.
I'm using the following command 
`dotnet linq2db scaffold -i c:\scaffold\config.json`
and the config.json file is defined like this:

```
{
    "general": {
        "provider": "PostgreSQL",
        "connection": "Server=localhost;Port=5432;Database=r19d;User Id=new_pg_db;Password=123456;Pooling=true;MinPoolSize=10;MaxPoolSize=100",
        "output": "c:\\scaffold\\gen",
        "overwrite": true
    },
	"schema" : {
		"database-in-name":false,
		"include-schemas":["new_pg_db"]
	},
	"dataModel": {
		"include-default-schema-name":false,
		"include-db-name":false,
		"partial-entities":true
	},
	"code": {
		"single-file":false
	}
}
```
This generates the db model, but it doesn't work the way I would expect:
1) All tables and views are generated in a one big file. I would expect (single-file:false above) many small files.
2) Is it possible to skip explicit schema definition inside of Table Attribute?
At the moment every Table attribute defines explicitly db schema in generated code, which I don't want:
``` C#
[Table("some_user_table", **Schema = "new_pg_db"**)]
public partial class SomeUserTab
```
I want omit Schema="new_pg_db" part in each and every table and view. 
It also overwrites schema in my connection, so I don't want always access new_pg_db,- I want to configure it later.
Am I doing something wrong?

**Comment by MaceWindu** [comment, marked as answer]
Nothing wrong. Looks like I forgot to add replacement for `DefaultSchema` option from old API. To workaround it you need to add schema load interceptor and clear schema there.

Check this link https://linq2db.github.io/articles/CLI.html#schema-load-interceptors

**Reply by MaceWindu** [reply]
Will add missing option in next release https://github.com/linq2db/linq2db/issues/3673

**Reply by febus** [reply]
Hi MaceWindu,
thank you for a quick response!
In the meantime, I tried a solution in linq2db code myself :)
file DataModelLoader.Entities.cs, line 90
```C#
	// call interceptor after entity model completely configured
	_interceptors.PreprocessEntity(_languageProvider.TypeParser, entity);
	**entity.Metadata.Name = new SqlObjectName(entity.@Class.Name!);**
```
I added only last line. Why should it be possible to define Schema in a Table attribute? Why would you ever want to define Schema there?

What is about "single-file" : false? Any fix for that one ( I didn't look at it yet in the codebase :) )

**Reply by MaceWindu** [reply]
single file generated, because non-default schema mappings generated in nested class. use this option to generate them as top-level classes:

```json
"dataModel": { "schema-as-type": false }
```

If you use `GetTables` interceptor I proposed, which works on schema load stage, instead of `PreprocessEntity`, which is executed later - you can ommit `schema-as-type` option as there will be no schema.

**Reply by febus** [reply]
Yes, I already created interceptor assembly. And yes, I've got immediately after that many small files - exactly how you described it.
Thank you for the explanation!

Is there  any way to rename generated Context class? (or how do you call it right? ) I mean this class, containing all other stuff ( tables and views etc)
```C#
namespace DataModel
{
	public partial class R19DDb : DataConnection
	{
		public R19DDb()
		{
			InitDataContext();
		} 
                ...
        }
}
```

I would like to name it differently, not R19DDb. R19d is the database name from the connection string. 
Is there any parameter or  interceptor to rename this generated class ?

**Reply by MaceWindu** [reply]
Yep, check `context-name` option

**Reply by MaceWindu** [reply]
New option will be in next release https://github.com/linq2db/linq2db/pull/3750
